include/alpaka/vec/Vec.hpp

```diff
@@ -402,6 +404,34 @@ namespace alpaka
             return r;
         }
 
+        //! \return The element-wise minimum of one or more vectors.
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<typename... Vecs, typename = std::enable_if_t<boost::mp11::mp_same<Vec, Vecs...>::value>>
+        ALPAKA_FN_HOST_ACC friend constexpr auto elementwise_min(Vec const& p, Vecs const&... qs) -> Vec
+        {
+            Vec r;
+            if constexpr(TDim::value > 0)
+            {
+                for(typename TDim::value_type i = 0; i < TDim::value; ++i)
+                    r[i] = std::min({p[i], qs[i]...});
+            }
+            return r;
+        }
+
+        //! \return The element-wise maximum of one or more vectors.
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<typename... Vecs, typename = std::enable_if_t<boost::mp11::mp_same<Vec, Vecs...>::value>>
+        ALPAKA_FN_HOST_ACC friend constexpr auto elementwise_max(Vec const& p, Vecs const&... qs) -> Vec
```

Comment:
Since you seem to intend to call this functions via a qualified name `alpaka::elementwise_max` and not ADL-only: please move it out of the definition of the `Vec` class and make it an ordinary free function. This should also make your unit test compile.

Reply:
Ah... I read the first part of

> A name first declared in a friend declaration within a class or class template `X` becomes a member of the innermost enclosing namespace of `X`, ...

but  I missed the second part

... but is not visible for lookup (except argument-dependent lookup that considers `X`) unless a matching declaration at namespace scope is provided.

Reply:
Ok, now you added declarations of the functions outside `Vec`. Why not move the entire definition outside as well? That would reduce the amount of code.

Reply:
I assumed they still need to be declared as `friend` to access the `Vec`'s members ?

Reply:
Yes they would, but IIUC you are not accessing anything private.